crates/runtime_isolate/src/bindings/compression.rs

```diff
@@ -0,0 +1,227 @@
+    GzEncoder(GzEncoder<Vec<u8>>),
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+enum CompressionFormat {
+    #[serde(rename = "gzip")]
+    Gz,
+    #[serde(rename = "deflate")]
+    Deflate,
+    #[serde(rename = "deflate-raw")]
+    DeflateRaw,
+}
+
+fn compression_create(
+    scope: &mut v8::HandleScope,
+    args: v8::FunctionCallbackArguments,
+) -> Result<String> {
+    let format: CompressionFormat = serde_v8::from_v8(scope, args.get(0))?;
+    let is_decoder: bool = serde_v8::from_v8(scope, args.get(1))?;
+    let w = Vec::new();
+
+    let inner = match format {
+        CompressionFormat::Gz => {
+            if is_decoder {
+                CompressionInner::GzDecoder(GzDecoder::new(w))
+            } else {
+                CompressionInner::GzEncoder(GzEncoder::new(w, Compression::default()))
+            }
+        }
+        CompressionFormat::Deflate => {
+            if is_decoder {
+                CompressionInner::DeflateDecoder(ZlibDecoder::new(w))
+            } else {
+                CompressionInner::DeflateEncoder(ZlibEncoder::new(w, Compression::default()))
+            }
+        }
+        CompressionFormat::DeflateRaw => {
+            if is_decoder {
+                CompressionInner::DeflateRawDecoder(DeflateDecoder::new(w))
+            } else {
+                CompressionInner::DeflateRawEncoder(DeflateEncoder::new(w, Compression::default()))
+            }
+        }
+    };
+
+    let id = Uuid::new_v4().to_string();
+
+    let isolate_state = Isolate::state(scope);
+    let mut state = isolate_state.borrow_mut();
+
+    let table = &mut state.compression_table;
+
+    table.insert(id.clone(), inner);
+
+    Ok(id)
+}
+
+pub fn compression_create_binding(
+    scope: &mut v8::HandleScope,
+    args: v8::FunctionCallbackArguments,
+    mut retval: v8::ReturnValue,
+) {
+    match compression_create(scope, args) {
+        Ok(res) => {
+            match serde_v8::to_v8(scope, res) {
+                Ok(v8_val) => {
+                    retval.set(v8_val.into());
+                }
+                Err(err) => {
+                    let exception = v8_exception(scope, err.to_string().as_str());
+                    scope.throw_exception(exception);
+                }
+            };
+        }
+        Err(err) => {
+            let exception = v8_exception(scope, err.to_string().as_str());
+            scope.throw_exception(exception);
+        }
+    }
+}
+
+fn compression_write(
+    scope: &mut v8::HandleScope,
+    args: v8::FunctionCallbackArguments,
+) -> Result<ZeroCopyBuf> {
+    let id: String = serde_v8::from_v8(scope, args.get(0))?;
+    let input: ZeroCopyBuf = serde_v8::from_v8(scope, args.get(1))?;
+
+    let isolate_state = Isolate::state(scope);
+    let mut state = isolate_state.borrow_mut();
+
+    let table = &mut state.compression_table;
+
+    let inner = table.get_mut(&id).unwrap();
+
+    let out: Vec<u8> = match inner {
+        CompressionInner::DeflateDecoder(d) => {
```

Comment:
Could we have a better naming for this variable? Probably something like `decoder`

Reply:
Because the `inner`'s type is CompressionInner.